appserver/appclient/client/acc/src/test/java/org/glassfish/appclient/client/CLIBootstrapTest.java

```diff
@@ -81,4 +87,50 @@ private void runTest(final String envVarName) {
         );
     }
 
+    @Test
+    public void testEscapingOfEscapedCommandLineArguments() throws Throwable {
+        Class CommandLineArgumentClass = Class.forName("org.glassfish.appclient.client.CLIBootstrap$CommandLineArgument");
+        Constructor constructor = CommandLineArgumentClass.getDeclaredConstructor(CLIBootstrap.class, String.class, int.class);
+        constructor.setAccessible(true);
+        Object commandLineArgument = constructor.newInstance(new CLIBootstrap(), ".*", Pattern.DOTALL);
+        Method formatMethod = CommandLineArgumentClass.getDeclaredMethod("format", StringBuilder.class, boolean.class, String.class);
+
+        assertAll(() -> {
+            String expect = "\"test\" \"append\"";
+            StringBuilder preCommandLineArgs = new StringBuilder("\"test\"");
+            String appendArg = "append";
+            StringBuilder actual = (StringBuilder) formatMethod.invoke(commandLineArgument, preCommandLineArgs, true, appendArg);
+            assertEquals(actual.toString(), expect);
+        }, () -> {
+            if (!OS.isWindows()) {
+                String expect = "\"a\\\\\\\"b\\\\\\$\\`c\\\"\\\\\"";
+                StringBuilder preCommandLineArgs = new StringBuilder("");
+                String appendArg = "a\\\"b\\$`c\"\\";
+                StringBuilder actual = (StringBuilder) formatMethod.invoke(commandLineArgument, preCommandLineArgs, true, appendArg);
+                assertEquals(actual.toString(), expect);
+            }
+        });
+    }
+
+    @Test
+    public void testAcceptabilityForEscapedDoubleQuotes() throws Throwable {
+        runTestUsingConvertInputArgsVariable("a\\\"b c\\\"d", new String[] { "a\\\"b", "c\\\"d" });
+    }
+
+    @Test
+    public void testAcceptabilityForQuotedEscapedDoubleQuotes() throws Throwable {
+        runTestUsingConvertInputArgsVariable("\"a \\\" b \\\" c\"", new String[] { "a \\\" b \\\" c" });
+    }
+
+    @Test
+    public void testAcceptabilityForQuotedEmptyStr() throws Throwable {
+        runTestUsingConvertInputArgsVariable("\"\"", new String[] { "" });
+    }
+
+    private void runTestUsingConvertInputArgsVariable( String inputArgs, String[] expect) throws Throwable {
+        Method convertInputArgsVariableMethod = CLIBootstrap.class.getDeclaredMethod("convertInputArgsVariable", String.class);
+        convertInputArgsVariableMethod.setAccessible(true);
+        String[] actual = (String[]) convertInputArgsVariableMethod.invoke(new CLIBootstrap(), inputArgs);
```

Comment:
```suggestion
        String[] actual = (String[]) convertInputArgsVariableMethod.invoke(null, inputArgs);
```
No need for instance.